pinot-broker/src/main/java/com/linkedin/pinot/broker/routing/builder/BalancedRandomRoutingTableBuilder.java

```diff
@@ -35,11 +36,13 @@
   private static final int DEFAULT_NUM_ROUTING_TABLES = 10;
   private static final String NUM_ROUTING_TABLES_KEY = "numOfRoutingTables";
 
+  private BrokerMetrics _brokerMetrics;
```

Comment:
Why not put this in base class?

Reply:
This would require me to either provide a setter or an init() method in the baseclass just for the purpose of setting this particular variable. It didn't seem like that added much in terms of readability, so decided to keep it in the subclasses where init is implemented.